### ISSUE TITLE
Docs for LambdaCallback should mention that it only works in training mode

### ISSUE THREAD

**Issue by DarkLight1337** [tensorflow/tensorflow#44256]
## URL(s) with the issue:

Documentation page for `tf.keras.callbacks.LambdaCallback`:
https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/LambdaCallback?hl=en

## Description of issue (what needs changing):

The docs for the base class `tf.keras.callbacks.Callback` mentions that the methods `on_epoch_begin`, `on_epoch_end`, `on_batch_begin` and `on_batch_end` are supposed to be called in training mode only, but this information is not included in the docs for `tf.keras.callbacks.LambdaCallback`. This may cause confusion when users pass this callback to the `callbacks` parameter of `tf.keras.Model.predict` or `tf.keras.Model.evaluate` and subsequently find out that the supplied functions are not being called.

I suggest that this information be added to the documentation to make it clear that `tf.keras.callbacks.LambdaCallback` does not do anything in predict and test mode.

Optionally, the functionality of `tf.keras.callbacks.LambdaCallback` could be expanded to include the rest of the available callback hooks in `tf.keras.callbacks.Callback` (`on_predict_batch_begin`, `on_predict_batch_end` etc.) so that it may be used in predict and test mode as well.

**Comment by AdeshKhandait** [comment]
I want to do.

**Comment by ymodak** [comment]
Thanks for your issue.
`tf.keras.callbacks.LambdaCallback` works with [`model.evaluate`](https://www.tensorflow.org/api_docs/python/tf/keras/Model?version=nightly#evaluate) and [`.predict`](https://www.tensorflow.org/api_docs/python/tf/keras/Model?version=nightly#predict) methods since they both take `batch_size` arg.
The docs are updated to add this information.
